src/uv-common.h

```diff
@@ -227,18 +228,20 @@ void uv__signal_cleanup(void);
 void uv__threadpool_cleanup(void);
 
 #define uv__has_active_reqs(loop)                                             \
-  ((loop)->active_reqs.count > 0)
+  (atomic_load((_Atomic unsigned int*) &((loop)->active_reqs.count)) > 0)
 
 #define uv__req_register(loop, req)                                           \
   do {                                                                        \
-    (loop)->active_reqs.count++;                                              \
+    atomic_fetch_add((_Atomic unsigned int*) &((loop)->active_reqs.count), 1);\
   }                                                                           \
   while (0)
 
+// TODO(trevnorris): This is technically a race condition since count could
+// change between the calls to assert() and fetch_sub().
 #define uv__req_unregister(loop, req)                                         \
   do {                                                                        \
     assert(uv__has_active_reqs(loop));                                        \
-    (loop)->active_reqs.count--;                                              \
+    atomic_fetch_sub((_Atomic unsigned int*) &((loop)->active_reqs.count), 1);\
```

Comment:
I think x86 will use “lock xadd” regardless for this

Reply:
As the last remaining Alpha user I feel offended by that statement.

<details>I jest, I sold my DEC desktop back in '02.</details>

Reply:
Most other platforms also have a faster relaxed add. But note that ‘lock add’ is also currently what modern compilers will emit for a full fence on x86, as the fastest full fence that exists. So the penalty here just potentially highest on x86.

Reply:
I just tried the following change, and the `loop_alive` performance dropped 10% on my i9:
```diff
diff --git a/src/uv-common.h b/src/uv-common.h
index 0786ec2b..467dd442 100644
--- a/src/uv-common.h
+++ b/src/uv-common.h
@@ -230,3 +230,3 @@ void uv__threadpool_cleanup(void);
 #define uv__has_active_reqs(loop)                                             \
-  (atomic_load((_Atomic unsigned int*) &((loop)->active_reqs.count)) > 0)
+  (uv__load_relaxed(&((loop)->active_reqs.count)) > 0)
 ```
Granted, it's the difference between 261289771/s and 231635107/s.

My biggest concern is what to do about the fact that Windows doesn't support C11 atomics.

Reply:
@bnoordhuis do you mean using `uv__load_relaxed()`? I'm actually confused there because when I compile with clang it uses the `#define uv__load_relaxed(p) (*p)`. Which doesn't do anything atomically anyways.

Reply:
I mean `atomic_load_explicit(ptr, memory_order_relaxed)`

Reply:
Awesome. Done.

Any ideas on how to address the fact that Windows doesn't support C11 atomics?